### PR TITLE
Changed mount syntax to be host-path:vm-path

### LINKED ISSUE #1344: Multiple `minikube mount` points, configurable
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Feature request

We should be able to configure the mount points with `minikube mount`, possibly with docker volume syntax like `minikube mount /Users:/hostdata` which would mount the host `/User` folder into the VM at `/hostdata`.

This shouldn't be a priority, as users shouldn't _really_ care where the data is mounted in the VM (since they can mount it in their pods at a different mount point), but might be useful to have different mounts.  

This might make the daemon lifecycle trickier, but would allow users to have multiple mounts and specific mount points.

### PR DESCRIPTION
**Pull request by aaron-prindle**
Fixes #1344

### CODE REVIEW

**Comment by naphthalene** [comment]
Nice!

**Comment by r2d4** [comment]
@minikube-bot retest this please

**Comment by aaron-prindle** [comment]
@minikube-bot retest this please

**Comment by aaron-prindle** [comment]
@minikube-bot retest this please

**Comment by aaron-prindle** [comment]
@minikube-bot retest this please

File: pkg/minikube/cluster/commands.go [hunk 1]
```diff
@@ -178,16 +178,30 @@ func GetLogsCommand(follow bool) (string, error) {
...
 }
 
-func GetMount9pCommand(ip net.IP) string {
-	return fmt.Sprintf(`
-sudo mkdir /mount-9p;
-sudo mount -t 9p -o trans=tcp -o port=5640 -o uid=1001 -o gid=1001 %s /mount-9p;
-sudo chmod 775 /mount-9p;`, ip)
+var mountTemplate = `
+sudo mkdir {{.Path}};
+sudo mount -t 9p -o trans=tcp -o port=5640 -o uid=1001 -o gid=1001 {{.IP}} {{.Path}};
```
Comment (r2d4) [inline pkg/minikube/cluster/commands.go]: Not exactly related to this PR, but do we need to change uid gid to be `$(id -u docker)`?

https://github.com/zchee/docker-machine-driver-xhyve/pull/171/files

File: pkg/minikube/cluster/commands.go [hunk 2]
```diff
@@ -178,16 +178,30 @@ func GetLogsCommand(follow bool) (string, error) {
...
+`, path, path)
 }
 
-func GetMount9pCommand(ip net.IP) string {
-	return fmt.Sprintf(`
-sudo mkdir /mount-9p;
-sudo mount -t 9p -o trans=tcp -o port=5640 -o uid=1001 -o gid=1001 %s /mount-9p;
-sudo chmod 775 /mount-9p;`, ip)
+var mountTemplate = `
+sudo mkdir {{.Path}};
```
Comment (r2d4) [inline pkg/minikube/cluster/commands.go]: Should we do a `mkdir -p` here?

File: cmd/minikube/cmd/mount.go [hunk 3]
```diff
@@ -36,42 +38,52 @@ var mountCmd = &cobra.Command{
...
+		mountString := args[0]
+		idx := strings.LastIndex(mountString, ":")
+		if idx == -1 { // no ":" was present
+			errText := fmt.Sprintf(`mount directory must be in the form: 
+			HOST_MOUNT_DIRECTORY:VM_MOUNT_DIRECTORY`)
+			fmt.Fprintln(os.Stderr, errText)
+			os.Exit(1)
+		}
+		hostPath := mountString[:idx]
+		vmPath := mountString[idx+1:]
```
Comment (r2d4) [inline cmd/minikube/cmd/mount.go]: I think you need to check for the cases where either hostPath or vmPath will be ""

`/c/Users:`
`:/home`
`:`

File: cmd/minikube/cmd/mount.go [hunk 4]
```diff
@@ -36,42 +38,52 @@ var mountCmd = &cobra.Command{
...
+\tminikube mount HOST_MOUNT_DIRECTORY:VM_MOUNT_DIRECTORY(ex:"/host-home:/vm-home")
 `
 			fmt.Fprintln(os.Stderr, errText)
 			os.Exit(1)
 		}
-		if _, err := os.Stat(args[0]); err != nil {
+		mountString := args[0]
+		idx := strings.LastIndex(mountString, ":")
+		if idx == -1 { // no ":" was present
+			errText := fmt.Sprintf(`mount directory must be in the form:
```
Comment (r2d4) [inline cmd/minikube/cmd/mount.go]: Maybe define errText as a const and use it in both places?

File: cmd/minikube/cmd/mount.go [hunk 5]
```diff
@@ -36,42 +38,52 @@ var mountCmd = &cobra.Command{
 	Long:  `Mounts the specified directory into minikube.`,
 	Run: func(cmd *cobra.Command, args []string) {
 		if len(args) != 1 {
-			errText := `Please specify a driver name and a directory to be mounted:
-\tminikube mount MOUNT_DIRECTORY(ex:"/home")
+			errText := `Please specify the directory to be mounted:
```
Comment (r2d4) [inline cmd/minikube/cmd/mount.go]: here's the other errText

File: pkg/minikube/cluster/commands.go (hunk 1 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/commands.go]: Currently this doesn't change the owner of the files but I will change it to be get the docker user.  In the future we can try to implement this in UFS.

File: cmd/minikube/cmd/mount.go (hunk 5 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/mount.go]: The errors are different and there are unique/more-verbose errors for each incorrect input

File: cmd/minikube/cmd/mount.go (hunk 3 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/mount.go]: I added check for vmPath, hostPath should already be g2g with the os.Stat check

**Pull request merged**